github-issue-summary-to-email/main.bal

```diff
@@ -0,0 +1,53 @@
+import ballerina/http;
+import ballerina/log;
+import ballerinax/github;
+import wso2/choreo.sendemail as email;
+
+// Github client configuration parameters
+configurable http:BearerTokenConfig gitHubTokenConfig = ?;
+configurable string repositoryName = ?;
+configurable string repositoryOwner = ?;
+
+// Email configuration parameters 
+configurable string recipientAddress = ?;
+
+public function main() returns error? {
+    string assigneeSummary = "";
+    github:Client githubClient = check new ({
+        auth: {
+            token: gitHubTokenConfig.token
+        }
+    });
+    stream<github:User,github:Error?> collaborators = check githubClient->getCollaborators(repositoryOwner, repositoryName);
+    log:printInfo("Started compiling the report");
+    check collaborators.forEach(function (github:User user ){
+        string query = string `repo:${repositoryOwner}/${repositoryName} is:issue assignee:${user.login}`;
+        github:SearchResult|github:Error issuesForAssignee = githubClient-> search(query, github:SEARCH_TYPE_ISSUE, 1);
+        if issuesForAssignee is github:SearchResult {
+            string userName = user?.name ?: "Unknown Name";
+            assigneeSummary += string `${userName} : ${issuesForAssignee.issueCount} ${"\n"}`;
+        } else {
+            log:printError("Error while searching issues of an assignee.",'error = issuesForAssignee);
+        }
+    });
+
+    string query = string `repo:${repositoryOwner}/${repositoryName} is:issue is:open`;
+    github:SearchResult|github:Error openIssues = githubClient-> search(query, github:SEARCH_TYPE_ISSUE, 1);
+    if openIssues is github:Error {
+        log:printError("Error while searching open issues.", 'error = openIssues);
+    }
+    int totalOpenIssueCount = openIssues is github:SearchResult? openIssues.issueCount : 0;
+    query = string `repo:${repositoryOwner}/${repositoryName} is:issue is:closed`;
+    github:SearchResult|github:Error closedIssues = githubClient-> search(query, github:SEARCH_TYPE_ISSUE, 1);
+    if closedIssues is github:Error {
+        log:printError("Error while searching closed issues.", 'error = closedIssues);
+    }
+    int totalClosedIssueCount = closedIssues is github:SearchResult? closedIssues.issueCount :0;
+
+    string issueSummary = string `ISSUE SUMMARY REPORT${"\n\n"}Repository Name: ${repositoryName}
+        ${"\n"}Total Issues Open: ${totalOpenIssueCount} ${"\n"}Total Issues Closed: ${totalClosedIssueCount}
+        ${"\n\n"}Issue Count by Assignee: ${"\n"}${assigneeSummary} ${"\n"}`;
+    email:Client emailClient = check new ();
+    string sendEmailResponse = check emailClient->sendEmail(recipientAddress, "Git Issue Summary", issueSummary);
+    log:printInfo("Email sent successfully \n " + sendEmailResponse);
+}
```

Comment:
Since the example is bit too wordy add a comment line before each major step.

Reply:
Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)